### ISSUE TITLE
Branch filtering on any jobs

### ISSUE THREAD

**Issue by jithine** [screwdriver-cd/screwdriver#1450]
## Context

Branch filtering applies only to trigger jobs with `~pr` or `~commit`. It would be nice to have this applied to a downstream job in the workflow so that certain paths can be activated based on the branch.

## Objective

job in `requires` to support branch filtering
  Example: 

```
build:
  requires: [~commit, ~commit:feature-1]
push-cloud:
  requires [build:master]
```

`push-cloud` will run only on commits against master branch
